packages/core/src/generators/utils/prompt-for-template.ts

```diff
@@ -0,0 +1,35 @@
+import { DotNetClient } from '@nx-dotnet/dotnet';
+
+import { prompt } from 'inquirer';
+
+export function promptForTemplate(
+  client: DotNetClient,
+  search?: string,
+  language?: string,
+): string | Promise<string> {
+  const available = client.listInstalledTemplates({ search, language });
+
+  if (available.length === 1) {
+    return available[0].shortNames[0];
+  }
+
+  if (search) {
+    const exactMatch = available.find((x) => x.shortNames.includes(search));
+    if (exactMatch) {
+      return exactMatch.shortNames[0];
+    }
+  }
+
+  return prompt<{ template: string }>([
+    {
+      name: 'template',
+      type: 'list',
+      message:
+        'What template should the project be initialized with? (https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-new#template-options) ',
```

Comment:
Users may expect additional help if their desired template isn't shown in this list. The page linked in the message briefly discusses installing new templates (though way above the anchor specified). Will that be enough of a description? I recognize there's not much room here to add more documentation.